src/utils.cc

```diff
@@ -574,6 +575,7 @@ std::string TemporaryFile::PathString() const { return Path().string(); }
 TemporaryDirectory::TemporaryDirectory(const std::string &hint)
     : tmp_name_(SafeTempRoot::Get() / boost::filesystem::unique_path("%%%%-%%%%-" + hint)) {
   boost::filesystem::create_directories(tmp_name_);
+  chmod(tmp_name_.c_str(), S_IRWXU);
```

Comment:
Again, is there not a way to create the dir and set the permissions all at once?